Modix.Services/Promotions/PromotionService.cs

```diff
@@ -116,56 +111,50 @@ public PromotionService(DiscordSocketClient client, IPromotionRepository reposit
 
         public void ThrowIfNotStaff(SocketGuildUser user)
         {
-            if (CurrentGuild.Owner == user)
-            {
-                return;
-            }
+            if (CurrentGuild.Owner == user) return;
 
-            if (!user.HasRole(_staffRoleId))
-            {
+            if (!user.HasRole(allowedToCreateRoleID))
                 throw new ArgumentException("The given promoter is not a staff member.");
-            }
         }
 
         public async Task<PromotionCampaign> CreateCampaign(SocketGuildUser user, string commentBody)
         {
-            if ((await GetCampaigns()).Any(d=>d.UserId == user.Id))
-            {
+            if (user == null)
+                throw new ArgumentNullException(nameof(user));
+            if (string.IsNullOrWhiteSpace(commentBody))
+                throw new ArgumentException("Comment cannot be empty!");
+            if ((await GetCampaigns()).Any(d => d.PromotionFor.DiscordUserId == user.Id))
                 throw new ArgumentException("A campaign already exists for that user.");
-            }
 
-            if (user.Roles.Count > 1)
-            {
-                throw new ArgumentException("Recommended user must be unranked.");
-            }
+            if (user.Roles.Count > 1) throw new ArgumentException("Recommended user must be unranked.");
 
             var timeOnServer = DateTimeOffset.UtcNow - user.JoinedAt.Value.ToUniversalTime();
 
             if (timeOnServer < TimeSpan.FromDays(30))
-            {
-                throw new ArgumentException($"Recommended user must have been a member of the server for more than 30 days. Currently: {timeOnServer.TotalDays}");
-            }
+                throw new ArgumentException(
+                    $"Recommended user must have been a member of the server for more than 30 days. Currently: {timeOnServer.TotalDays}");
 
             var ret = new PromotionCampaign
             {
-                UserId = user.Id,
-                Username = $"{user.Username}#{user.Discriminator}",
                 StartDate = DateTimeOffset.UtcNow,
                 Status = CampaignStatus.Active
             };
 
+            await _repository.AddCampaign(ret, user);
+
             await AddComment(ret, commentBody, PromotionSentiment.For);
 
-            await _repository.AddCampaign(ret);
+            if (PromotionChannel == null)
+                throw new NullReferenceException(nameof(promotionChannelID));
 
-            await PromotionChannel?.SendMessageAsync("", false, 
+            await PromotionChannel?.SendMessageAsync("", false,
```

Comment:
As earlier, I think the safe derefencing can be removed: `PromotionChannel?.SendMessageAsync` -> `PromotionChannel.SendMessageAsync`.